packages/edit-site/src/components/sidebar/global-styles-v2/screens/colors-palette.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	CardBody,
+	ColorCircle,
+	HStack,
+	ListGroup,
+	ListGroupHeader,
+	ListGroups,
+	SegmentedControl,
+	VStack,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { Screen, ScreenHeader } from '../components';
+import { useAppState } from '../state';
+
+const Palette = ( { colors = [], title, index } ) => {
+	const appState = useAppState();
+
+	const handleOnClick = ( i ) => () => {
+		appState.setColorPickerKey(
+			`color.palettes[${ index }].colors[${ i }].color`
```

Comment:
Again, we'll need to figure out a dev-friendly way to do things like this. I think it's a good idea to work on the prototype of the state in a separate PR.

Reply:
We already have getStyle setStyle etc abstractions on packages/edit-site/src/components/editor/global-styles-provider.js.
This logic is complex e.g: when the user sets a color if the color being set is a theme color or user color the color should be referenced using a CSS var instead of the value. When the value is for example CSS custom variable the components should retrieve the value for that value and show it.
Our getters and setters try to abstract everything from developers.